### DISCUSSION TITLE
convert vs. magick

### DISCUSSION THREAD

**Discussion by RexBlock** [ImageMagick/ImageMagick#7381] [Development] [answered]
### ImageMagick version

7.1.1-33

### Operating system, version and so on

MacOS 12.6.5 (Monterey) M1

### Description

I use some scripts that were written by Fred Weinhaus ([link](http://www.fmwconcepts.com/imagemagick/index.php)) to do some fun stuff with images, specifically putting frames around the images. A lot of Fred's scripts use the 'convert' command, which have recently started throwing the deprecation warning after I did an IM upgrade. 

I don't know if Fred's scripts are supported any more (have not yet reached out to him), and some of them have not been updated in a long time. I've been using them unchanged since 2018. 

To get through this issue, the first question I have is in the syntax of 'convert' vs that of 'magick', and basically how different they are. I took one script ([bordereffects](http://www.fmwconcepts.com/imagemagick/bordereffects/index.php)) and simply changed /convert/magick/g. 

So far it looks ok, but how much can I rely on syntax in calls to 'convert' being the same as to 'magick'? 

Thanks for your help.

**Comment by fmw42** [comment]
My script will use IM 7, if you configure your system with a symbolic link between convert and magick. 

This can be done also by copying the magick file to a convert file in the utilities directory

```
cd
cd /full_path_to_IM7_download/ImageMagick-7.1.1-33/utilities/
cp magick convert
cd

```

It is possible that some scripts have issues or differences under IM 7. If so, contact me and I will make sure they are updated for IM 7 differences.

Sorry, I will amend my home page to give these details.

**Reply by RexBlock** [reply, marked as answer]
Thank you Fred, that does seem to be the way to go. I'll keep an eye out for anywhere else it might be happening and if there aren't other hidden issues creeping in. 

Thanks again for all the work you put into those. They really do demonstrate the vast potential of magick.
